### PR TITLE
Add `rayleigh` distribution to `random.py`

### PR DESCRIPTION
**Pull request by JiaYaobo**
Add a counterpart  of `numpy.random.rayleigh` for JAX

### CODE REVIEW

File: jax/_src/random.py [hunk 1]
```diff
@@ -1828,3 +1828,46 @@ def ball(
   g = generalized_normal(k1, p, (*shape, d), dtype)
   e = exponential(k2, shape, dtype)
   return g / (((jnp.abs(g) ** p).sum(-1) + e) ** (1 / p))[..., None]
+
+
+def rayleigh(key: KeyArray,
+              scale: RealArray,
+              shape: Optional[Shape] = None,
+              dtype: DTypeLikeFloat = dtypes.float_) -> Array:
```
Comment (sharadmv) [inline jax/_src/random.py:1836]: Please fix indentation here.

**Review by sharadmv — APPROVED** [review]
LGTM other than indentation issue

File: jax/_src/random.py (hunk 1 above)
Comment (JiaYaobo) [reply jax/_src/random.py]: Thanks for your review, modification is committed

File: jax/_src/random.py [hunk 2]
```diff
@@ -1828,3 +1828,46 @@ def ball(
...
+@partial(jit, static_argnums=(2, 3), inline=True)
+def _rayleigh(key, scale, shape, dtype) -> Array:
+  if shape is None:
+    shape = np.shape(scale)
+  else:
+    _check_shape("rayleigh", shape, np.shape(scale))
+  u = uniform(key, shape, dtype)
+  log_u = lax.log(u)
+  n_two = _lax_const(scale, -2)
+  sqrt_u = lax.sqrt(lax.mul(log_u, n_two))
```
[the remark this answers is not in the document]
Comment (JiaYaobo) [reply jax/_src/random.py:1871]: Thanks for your review!

**Review by jakevdp — APPROVED** [review]

**Comment by JiaYaobo** [comment]
@jakevdp Hi, the workflow indicates the check `import/copybara: Pending —Waiting for internal safe review approval`, do I need to do something else to activate this check?

**Comment by sharadmv** [comment]
Sorry, I can take care of that!

**Pull request merged**